### PR TITLE
serializes the input blocks when sending to RPCs Fixes #1278

### LINKED ISSUE #1278: Cannot send a custom block in an rpc call
rpc-fn-call

These blocks should be serialized (xml) and then sent to the server.

This will be necessary for #1152

### PR DESCRIPTION
**Pull request by hamidzr**
I'm not sure if `.toBlockXML` is the right call. 
`ActionManager.serialize` ?

### CODE REVIEW

File: src/client/threads.js [hunk 1]
```diff
@@ -341,6 +341,11 @@ NetsProcess.prototype.getJSFromRPCStruct = function (rpc, methodSignature) {
         if (values[index] instanceof List) {
             return toQueryString(values[index], name);
```
Comment (brollb) [inline src/client/threads.js]: what if the list contains blocks?

**Comment by hamidzr** [comment]
I changed it to use the serializer from snapActions.
>> Regardless, we will have to be careful because the serializers will cache duplicates 

 if the serializers don't share the same cache, can we eliminate the problem by having a separate serializer? 
What does a serializer cache by? (the key)

**Comment by brollb** [comment]
https://github.com/brollb/snap-undo/blob/2b46d0d37f6d8b039871329feb17483540fbcfe0/store.js#L279-L280

File: src/client/threads.js [hunk 2]
```diff
@@ -319,12 +319,18 @@ NetsProcess.prototype.parseRPCResult = function (result) {
...
+    return blockXML;
+}
+
 function toQueryString(list, prefix) {
     var array = list.contents;
     var str = [], k, v;
     for(var i = 0; i < array.length; i++) {
         k = prefix + '[' + i + ']';
-        v = array[i];
+        v = array[i] instanceof Context ? blockToText(array[i]) : array[i];
```
Comment (brollb) [inline src/client/threads.js]: use `SnapActions.serializeBlock(array[i], true)` instead

File: src/client/threads.js [hunk 3]
```diff
@@ -319,12 +319,18 @@ NetsProcess.prototype.parseRPCResult = function (result) {
     return result;
 };
 
+
+function blockToText(block) {
```
Comment (brollb) [inline src/client/threads.js]: remove this in favor of the below comment

**Comment by hamidzr** [comment]
so when I pass in a custom block, should it also serialize the blocks used in creating that block?!

**Pull request closed without merging**